modules/overrides/userconfigurable/api/limits_test.go

```diff
@@ -0,0 +1,119 @@
+package api
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/grafana/tempo/modules/overrides"
+	filterconfig "github.com/grafana/tempo/pkg/spanfilter/config"
+)
+
+func Test_limitsFromOverrides(t *testing.T) {
+	userID := "foo"
+
+	cfg := overrides.Config{
+		Defaults: overrides.Overrides{
+			Forwarders: []string{"my-forwarder"},
+			MetricsGenerator: overrides.MetricsGeneratorOverrides{
+				Processors:         map[string]struct{}{"service-graphs": {}},
+				CollectionInterval: 15 * time.Second,
+				DisableCollection:  true,
+				Processor: overrides.ProcessorOverrides{
+					ServiceGraphs: overrides.ServiceGraphsOverrides{
+						HistogramBuckets:         []float64{0.1, 0.2, 0.5},
+						Dimensions:               []string{"my-dim1", "my-dim2"},
+						PeerAttributes:           []string{"db.name"},
+						EnableClientServerPrefix: true,
+					},
+					SpanMetrics: overrides.SpanMetricsOverrides{
+						Dimensions:       []string{"your-dim1", "your-dim2"},
+						EnableTargetInfo: true,
+						FilterPolicies: []filterconfig.FilterPolicy{
+							{
+								Exclude: &filterconfig.PolicyMatch{
+									MatchType: filterconfig.Regex,
+									Attributes: []filterconfig.MatchPolicyAttribute{
+										{
+											Key:   "resource.service.name",
+											Value: "unknown_service:myservice",
+										},
+									},
+								},
+							},
+						},
+						HistogramBuckets:             []float64{1, 2, 5},
+						TargetInfoExcludedDimensions: []string{"no"},
+					},
+				},
+			},
+		},
+	}
+	overridesInt, err := overrides.NewOverrides(cfg)
+	assert.NoError(t, err)
+
+	limits := limitsFromOverrides(overridesInt, userID)
+	limitsJson, err := json.MarshalIndent(limits, "", "  ")
+	assert.NoError(t, err)
+
+	expectedJson := `{
```

Comment:
Does this test pick up if we forget to add a new field to `limitsFromOverrides` ?  I think it does since the output json will get a null but just checking.

Reply:
It does because `overrides.MetricsGeneratorOverrides` will set the new value to the zero value which will be omitted in the json and you get a diff.

This what happens if I remove collection_interval from both the input overrides and the expected result:

```
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -9,2 +9,3 @@
        	            	     "disable_collection": true,
        	            	+    "collection_interval": "0s",
        	            	     "processor": {
        	Test:       	Test_limitsFromOverrides
--- FAIL: Test_limitsFromOverrides (0.00s)
```